students/jeff_shabani/session05/mailroom_3.py

```diff
@@ -0,0 +1,170 @@
+#!/usr/bin/env python3
+import os
+import sys
+
+from collections import OrderedDict
+from operator import itemgetter
+from pathlib import Path
+
+DONORS = {'William B': [120, 130, 50],
+          'Sammy Maudlin': [500, 125, 670, 1000],
+          'Bobby Bittman': [10],
+          'Skip Bittman': [75, 125, 19],
+          'Ashley Lashbrooke': [10000, 15000]}
+
+prompt = "\n".join(("Welcome to my charity!",
+                    "Please select and option below:",
+                    "1 - Send a Thank You to an individual",
+                    "2 - Create a Report",
+                    "3 - Send letters to all donors",
+                    "4 - Quit",
+                    ">>> "))
+
+def value_error():
+    """
+    Catch non-numeric entries for donation amount.
+    """
+    amount = int()
+    while not amount:
+        try:
+            amount = int(input('How much would this donor like to donate?'))
+        except ValueError:
+            print('Please enter a valid numerical amount.')
+        else:
+            return amount
+
+
+def view_donor_names():
+    for name in DONORS:
+        print(name)
+
+
+def write_a_letter(name, amount):
+    letter = f'Dear {name},\n\nThank you for your kind donation of ${amount:,.0f}\n\n' \
+        f'Rest assured that these funds will be put to optimal use.\n\n' \
+        f'Best regards,\n' \
+        f'The Charitable Charities Team'
+    return letter
+
+
+def create_directory_decision():
+    """
+    Ask user if they want the directory they entered
+    created if it does not currently exsit.
+    """
+    answer = input('Path does\'t exist. Do you want to create it?')
+    return answer
+
+
+def validate_letter_directory_path():
+    """
+    Check if user-entered directory exists and offer them the
+    choice to create it if not. Default is current working
+    directory
+    """
+    location = input(f'Please enter the full path of the directory\n'
+                     f'where you want to save your letters.\n'
+                     f'Hit <Enter> to save to the current working directory.')
+
+    if not location:
+        location = os.getcwd()
+    else:
+        location = Path(f'{location}')
+        if not location.exists():
+            create_directory_answer = create_directory_decision()
```

Comment:
There is no reason for this to be function call as it doens't do replace more than a single line
```
create_directory_answer = input('Path does\'t exist. Do you want to create it?')
```
is all you need.